### PR TITLE
Avoid spurious changes with bastion hosts due to user data

### PR DESCRIPTION
**Pull request by hakman**
Fixes: #13320 #14195

### CODE REVIEW

File: pkg/model/bootstrapscript.go [hunk 1]
```diff
@@ -245,7 +245,7 @@ func (b *BootstrapScriptBuilder) ResourceNodeUp(c *fi.ModelBuilderContext, ig *k
 
 		// Bastions can have AdditionalUserData, but if there isn't any skip this part
 		if len(ig.Spec.AdditionalUserData) == 0 {
-			return fi.NewStringResource(""), nil
+			return nil, nil
```
Comment (seh) [inline pkg/model/bootstrapscript.go:248]: I see that this avoids returning a resource, and that we wind up setting the "UserData" field of the `awstasks.LaunchTemplate` to nil. How does that change avoid creation of a new launch template. Is it that two different `*fi.StringResource` values don't compare as equal, so _kOps_ thinks that there's a material change required in the launch template?

Comment (hakman) [reply pkg/model/bootstrapscript.go:248]: AWS seems to return `nil` when `UserData == ""`. This means that kOps tries to reconcile it.

**Comment by seh** [comment]
Testing this patch atop the "v1.24.3" tag, I found that it resolves the problem, such that _kops update cluster_ no longer tries to create a new launch template version for the bastion ASG on every invocation.

**Review by olemarkus — APPROVED** [review]

**Pull request merged**